Filebeat panic at startup

While running beats-tester there is a panic at startup for Filebeat.

For confirmed bugs, please report:
- Version: 6.2.0 rc1
- Operating System: Window 2012 R2 
- Steps to Reproduce: Run Beats-Tester

```
2018-01-22T16:55:39.473-0800    INFO    prospector/prospector.go:138    Stopping Prospector: 15026298159737792444
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x40131c]

goroutine 29 [running]:
sync/atomic.AddUint64(0x13b043b4, 0x1, 0x0, 0x8f6be5, 0x2e)
        /usr/local/go/src/sync/atomic/asm_386.s:112 +0xc
github.com/elastic/beats/libbeat/common/atomic.(*Uint64).Add(0x13b043b4, 0x1, 0x0, 0x0, 0x145e940)
        /go/src/github.com/elastic/beats/libbeat/common/atomic/atomic.go:66 +0x33
github.com/elastic/beats/libbeat/common/atomic.(*Uint64).Inc(0x13b043b4, 0x184ac8, 0x0)
        /go/src/github.com/elastic/beats/libbeat/common/atomic/atomic.go:68 +0x31
github.com/elastic/beats/filebeat/prospector/log.(*Prospector).startHarvester(0x13b042d0, 0x0, 0x0, 0x0, 0x13a15c0, 0x13
af23c0, 0x13ac1340, 0x10, 0x0, 0x0, ...)
        /go/src/github.com/elastic/beats/filebeat/prospector/log/prospector.go:594 +0x3e
github.com/elastic/beats/filebeat/prospector/log.(*Prospector).scan(0x13b042d0)
        /go/src/github.com/elastic/beats/filebeat/prospector/log/prospector.go:435 +0x4e4
github.com/elastic/beats/filebeat/prospector/log.(*Prospector).Run(0x13b042d0)
        /go/src/github.com/elastic/beats/filebeat/prospector/log/prospector.go:162 +0x8c
github.com/elastic/beats/filebeat/prospector.(*Prospector).Run(0x13af9140)
        /go/src/github.com/elastic/beats/filebeat/prospector/prospector.go:111 +0x2b
github.com/elastic/beats/filebeat/prospector.(*Prospector).Start.func1(0x13ac4c60, 0x13af9140)
        /go/src/github.com/elastic/beats/filebeat/prospector/prospector.go:104 +0x4b
created by github.com/elastic/beats/filebeat/prospector.(*Prospector).Start
        /go/src/github.com/elastic/beats/filebeat/prospector/prospector.go:97 +0x13a
PS C:\Users\vagrant\filebeat-6.2.0-windows-x86>
```

[filebeat.log.txt](https://github.com/elastic/beats/files/1654124/filebeat.log.txt)
[filebeat.yml.txt](https://github.com/elastic/beats/files/1654126/filebeat.yml.txt)

This is also happening on ubuntu-precise-32 with config:

```
filebeat:
  prospectors:
    -
      paths:
        - /var/log/test.log

output:
  file:
    path: /root/output
    filename: filebeat
```

Log:
<details>
root@vagrant-ubuntu-precise-32:~# filebeat -e -d "*"
2018-01-23T05:00:58.498Z	INFO	instance/beat.go:468	Home path: [/usr/share/filebeat] Config path: [/etc/filebeat] Data path: [/var/lib/filebeat] Logs path: [/var/log/filebeat]
2018-01-23T05:00:58.499Z	DEBUG	[beat]	instance/beat.go:495	Beat metadata path: /var/lib/filebeat/meta.json
2018-01-23T05:00:58.501Z	INFO	instance/beat.go:475	Beat UUID: 1d097697-d99a-4352-b82a-2c3b3795782a
2018-01-23T05:00:58.502Z	INFO	instance/beat.go:213	Setup Beat: filebeat; Version: 6.2.0
2018-01-23T05:00:58.503Z	DEBUG	[beat]	instance/beat.go:230	Initializing output plugins
2018-01-23T05:00:58.503Z	DEBUG	[processors]	processors/processor.go:49	Processors: 
2018-01-23T05:00:58.503Z	INFO	fileout/file.go:76	Initialized file output. path=/root/output/filebeat max_size_bytes=10485760 max_backups=7 permissions=-rw-------
2018-01-23T05:00:58.504Z	INFO	pipeline/module.go:76	Beat name: vagrant-ubuntu-precise-32
2018-01-23T05:00:58.504Z	INFO	instance/beat.go:301	filebeat start running.
2018-01-23T05:00:58.504Z	DEBUG	[registrar]	registrar/registrar.go:88	Registry file set to: /var/lib/filebeat/registry
2018-01-23T05:00:58.505Z	INFO	registrar/registrar.go:108	Loading registrar data from /var/lib/filebeat/registry
2018-01-23T05:00:58.505Z	INFO	registrar/registrar.go:119	States Loaded from registrar: 0
2018-01-23T05:00:58.505Z	WARN	beater/filebeat.go:261	Filebeat is unable to load the Ingest Node pipelines for the configured modules because the Elasticsearch output is not configured/enabled. If you have already loaded the Ingest Node pipelines or are using Logstash pipelines, you can ignore this warning.
2018-01-23T05:00:58.505Z	INFO	crawler/crawler.go:48	Loading Prospectors: 1
2018-01-23T05:00:58.505Z	DEBUG	[processors]	processors/processor.go:49	Processors: 
2018-01-23T05:00:58.505Z	DEBUG	[prospector]	log/config.go:178	recursive glob enabled
2018-01-23T05:00:58.504Z	INFO	[monitoring]	log/log.go:95	Starting metrics logging every 30s
2018-01-23T05:00:58.506Z	DEBUG	[registrar]	registrar/registrar.go:150	Starting Registrar
2018-01-23T05:00:58.506Z	DEBUG	[prospector]	log/prospector.go:120	exclude_files: []. Number of stats: 0
2018-01-23T05:00:58.506Z	DEBUG	[prospector]	log/prospector.go:141	Prospector with previous states loaded: 0
2018-01-23T05:00:58.506Z	INFO	log/prospector.go:111	Configured paths: [/var/log/test.log]
2018-01-23T05:00:58.506Z	DEBUG	[prospector]	prospector/prospector.go:87	Starting prospector of type: log; ID: 14439312048545787300
2018-01-23T05:00:58.506Z	INFO	crawler/crawler.go:82	Loading and starting Prospectors completed. Enabled prospectors: 1
2018-01-23T05:00:58.506Z	DEBUG	[prospector]	log/prospector.go:147	Start next scan
2018-01-23T05:00:58.506Z	DEBUG	[prospector]	log/prospector.go:361	Check file for harvesting: /var/log/test.log
2018-01-23T05:00:58.506Z	DEBUG	[prospector]	log/prospector.go:434	Start harvester for new file: /var/log/test.log
2018-01-23T05:00:58.506Z	INFO	prospector/prospector.go:138	Stopping Prospector: 14439312048545787300
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x8662b0c]

goroutine 33 [running]:
sync/atomic.AddUint64(0x1a382594, 0x1, 0x0, 0x0, 0x1)
	/usr/local/go/src/sync/atomic/asm_386.s:112 +0xc
github.com/elastic/beats/libbeat/common/atomic.(*Uint64).Add(0x1a382594, 0x1, 0x0, 0x1a33b9c8, 0x1a54f7a0)
	/go/src/github.com/elastic/beats/libbeat/common/atomic/atomic.go:66 +0x33
github.com/elastic/beats/libbeat/common/atomic.(*Uint64).Inc(0x1a382594, 0xff, 0x9e34a79a)
	/go/src/github.com/elastic/beats/libbeat/common/atomic/atomic.go:68 +0x31
github.com/elastic/beats/filebeat/prospector/log.(*Prospector).startHarvester(0x1a3824b0, 0x0, 0x0, 0x0, 0x95dcae0, 0x1a5a0e10, 0x1a4e8fa0, 0x11, 0x0, 0x0, ...)
	/go/src/github.com/elastic/beats/filebeat/prospector/log/prospector.go:594 +0x3b
github.com/elastic/beats/filebeat/prospector/log.(*Prospector).scan(0x1a3824b0)
	/go/src/github.com/elastic/beats/filebeat/prospector/log/prospector.go:435 +0x4e4
github.com/elastic/beats/filebeat/prospector/log.(*Prospector).Run(0x1a3824b0)
	/go/src/github.com/elastic/beats/filebeat/prospector/log/prospector.go:162 +0x8c
github.com/elastic/beats/filebeat/prospector.(*Prospector).Run(0x1a576ac0)
	/go/src/github.com/elastic/beats/filebeat/prospector/prospector.go:111 +0x2b
github.com/elastic/beats/filebeat/prospector.(*Prospector).Start.func1(0x1a527dc0, 0x1a576ac0)
	/go/src/github.com/elastic/beats/filebeat/prospector/prospector.go:104 +0x4b
created by github.com/elastic/beats/filebeat/prospector.(*Prospector).Start
	/go/src/github.com/elastic/beats/filebeat/prospector/prospector.go:97 +0x13a
</details>

--- a/filebeat/input/log/config.go
+++ b/filebeat/input/log/config.go
@@ -67,7 +67,7 @@ type config struct {
 	Paths          []string        `config:"paths"`
 	ScanFrequency  time.Duration   `config:"scan_frequency" validate:"min=0,nonzero"`
 	CleanRemoved   bool            `config:"clean_removed"`
-	HarvesterLimit uint64          `config:"harvester_limit" validate:"min=0"`
+	HarvesterLimit uint32          `config:"harvester_limit" validate:"min=0"`
 	Symlinks       bool            `config:"symlinks"`
 	TailFiles      bool            `config:"tail_files"`
 	RecursiveGlob  bool            `config:"recursive_glob.enabled"`

--- a/filebeat/input/log/prospector.go
+++ b/filebeat/input/log/prospector.go
@@ -46,7 +46,7 @@ type Prospector struct {
 	outlet        channel.Outleter
 	stateOutlet   channel.Outleter
 	done          chan struct{}
-	numHarvesters atomic.Uint64
+	numHarvesters atomic.Uint32
 }
 
 // NewProspector instantiates a new Log

